### PR TITLE
Use a goroutine to block for the signal

### PR DESCRIPTION
**Pull request by mestery**
Experimenting with a more go-friendly way of handling signals.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: cmd/nsm/main.go [hunk 1]
```diff
@@ -20,10 +20,9 @@ func main() {
 		syscall.SIGINT,
 		syscall.SIGTERM,
 		syscall.SIGQUIT)
-	select {
-	case s := <-sigChan:
+	go func() {
```
Comment (edwarnicke) [inline cmd/nsm/main.go]: Is this still producing a race condition?

Comment (fkautz) [reply cmd/nsm/main.go]: What are we trying to accomplish here?

Comment (fkautz) [reply cmd/nsm/main.go]: Try using this to block main from exiting.

https://golang.org/pkg/sync/#WaitGroup

Comment (mestery) [reply cmd/nsm/main.go]: OK, I've got an update which adds WaitGroup support, and when coupled with #29, this now works as expected! New commit pushing in 30 seconds!

File: nsmdp/nsmdp.go [hunk 2]
```diff
@@ -36,6 +36,7 @@ func NewNSMDevicePlugin() *NSMDevicePlugin {
 		DevicePlugin: deviceplugin.NewDevicePlugin(serverSock, resourceName),
 		devs:         make(map[string]*NSMDevice),
 		stop:         make(chan interface{}),
+		updatedevs:   make(chan *NSMDevice, 10),
```
Comment (mestery) [inline nsmdp/nsmdp.go]: This is the change in #29, if we merge these two together, I'll just close #29. Or, I can remove this and we can merge them separately.

**Pull request merged**